servicenow/stackstate_checks/servicenow/servicenow.py

```diff
@@ -174,62 +160,59 @@ def _process_components(self, components, instance_config):
                 identifiers.append("urn:host:/{}".format(comp_name))
             identifiers.append(external_id)
             data.update(component)
-            data.update({"identifiers": identifiers, "tags": instance_config.instance_tags})
+            data.update({"identifiers": identifiers, "tags": instance_info.instance_tags})
 
             self.component(external_id, comp_type, data)
 
-    def _collect_relation_types(self, instance_config):
+    def _collect_relation_types(self, instance_info):
         """
         collects relations from CMDB
         :return: dict, raw response from CMDB
         """
+        auth = (instance_info.user, instance_info.password)
+        url = instance_info.url + '/api/now/table/cmdb_rel_type?sysparm_fields=sys_id,parent_descriptor'
+        return self._get_json(url, instance_info.timeout, auth)
 
-        base_url = instance_config.base_url
-        auth = instance_config.auth
-        url = base_url + '/api/now/table/cmdb_rel_type?sysparm_fields=sys_id,parent_descriptor'
-
-        return self._get_json(url, instance_config.timeout, auth)
-
-    def _process_relation_types(self, instance_config):
+    def _process_relation_types(self, instance_info):
         """
         collect available relations from cmdb_rel_ci
-        :return: nothing
         """
         relation_types = {}
-        state = self._collect_relation_types(instance_config)
+        types = self._collect_relation_types(instance_info)
 
-        if "result" in state:
-            for relation in state.get('result', []):
-                id = relation['sys_id']
+        if "result" in types:
+            for relation in types.get('result', []):
+                sys_id = relation['sys_id']
                 parent_descriptor = relation['parent_descriptor']
-                relation_types[id] = parent_descriptor
+                relation_types[sys_id] = parent_descriptor
```

Comment:
I know it's nitpicking, but these can potentially fail if it's not there. We can introduce schematics here as well to avoid the `if x in relation` stuff. I would then propose to do it at a root level starting with `types`. One thing to note though, we want to stay resilient so we'll have to catch the validation exception, log it and continue with an empty relation_types dict.

Reply:
You could argue that `_collect_relation_types` also just does the schematics validation and returns the list.

Reply:
Just remember to set `strict=False` when instantiating the schematics model like:
```
_event = Event(event, strict=False)  # strict=False ignores extra fields provided by nagios
_event.validate()
```
Then it will ignore extra fields and validation will be ok